src/lib.rs

```diff
@@ -150,7 +150,26 @@ impl<P> Hierarchy<P> {
 
     /// Get the children of a specific entity.
     pub fn children(&self, entity: Entity) -> &[Entity] {
-        self.children.get(&entity).map(|vec| vec.as_slice()).unwrap_or(&[])
+        self.children
+            .get(&entity)
+            .map(|vec| vec.as_slice())
+            .unwrap_or(&[])
+    }
+
+    /// Get all children in a sub hierarchy as a `BitSet`
+    pub fn all_children(&self, entity: Entity) -> BitSet {
```

Comment:
From a usability standpoint, getting an iterator is better than getting a bitset that I need to check against the hierarchy

Reply:
I disagree, typically I want this because I'm about to change some components on the entities.  This is easy to join into some component storages so I can make those changes.

Reply:
Can you add a method that returns an iterator of childs and intenally uses your all_children method?

Reply:
Sure, but out of curiosity what would you use it for?

Reply:
I don't know, but if we need it for a reason or another, we'll have it.

Reply:
Ok, well I just finished coding the method that used the current method internally and I hate the performance of it, so I'm changing it to use @Rhuagh 's iterator suggestion instead.  It still returns the same stuff.